### ISSUE TITLE
[Bug]: Incorrectly removed "unused" alias

### ISSUE THREAD

**Issue by foxfriends** [quarylabs/sqruff#2383]
### What happened?

Using PostgreSQL, this query is valid:

```sql
INSERT INTO some_table (other_id, field_value)
SELECT other_table.id, value
FROM other_table, (VALUES (1), (2)) AS field_values (value);
```

But when I run `sqruff fix`, the updated query is invalid because the table alias was detected as not used (the table name alias was not used, but the named column alias WAS used):

```sql
INSERT INTO some_table (other_id, field_value)
SELECT other_table.id, value
FROM other_table, (VALUES (1), (2)); -- NOTE: Missing alias
```

This problem can be avoided by explicitly using the aliased table name:

```sql
INSERT INTO some_table (other_id, field_value)
SELECT other_table.id, field_values.value
FROM other_table, (VALUES (1), (2)) as field_values (value);
```

I believe sqruff should detect the column alias usage and not remove the "unused" alias in this situation, or suggest adding the explicit usage of the table alias, rather than outputting a query that is invalid and not equivalent to the input.

### Version

0.34.1

### Checked in Playground

- [x] I have checked this bug in the [playground](https://playground.quary.dev/?secondary=Format).
